### PR TITLE
[CSL-2328] no reflection configurations for diffusion layer (part 1)

### PR DESCRIPTION
**Pull request by avieth**
## Description

Subset of #2590 intended to make it easier for people to review.

## Linked issue

[CSL-2328]

## Type of change
- [x] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

### CODE REVIEW

**Comment by avieth** [comment]
@karknu @coot perhaps you will find this one easier to review. It's part 1 of a bigger change

**Comment by erikd** [comment]
@avieth Please feel free to add me as a reviewer (assuming that you can).

I can definitely see the merit in these changes, but the PR is huge. I'm currently going through each separate commit in split diff mode. Its the only way to make sense of what is going on, but its time consuming.

File: ssc/Pos/Arbitrary/Ssc.hs [hunk 1]
```diff
@@ -1,8 +1,9 @@
 -- | Arbitrary instances and generators for SSC types.
 
 module Pos.Arbitrary.Ssc
-       ( SscPayloadDependsOnSlot (..)
-       , BadCommAndOpening (..)
+       ( {-SscPayloadDependsOnSlot (..)
+       , -}
```
Comment (erikd) [inline ssc/Pos/Arbitrary/Ssc.hs]: If that's not needed its probably better to drop it.

File: ssc/Pos/Arbitrary/Ssc.hs [hunk 2]
```diff
@@ -38,9 +38,11 @@ import           Pos.Util.QuickCheck.Arbitrary (Nonrepeating (..), sublistN)
 -- Types
 ----------------------------------------------------------------------------
 
+{-
 newtype SscPayloadDependsOnSlot = SscPayloadDependsOnSlot
     { genPayloadDependsOnSlot :: SlotId -> Gen SscPayload
     } deriving Generic
+-}
```
Comment (erikd) [inline ssc/Pos/Arbitrary/Ssc.hs]: Like the export above. Should either delete or uncomment it and actually export it.

File: ssc/Pos/Arbitrary/Ssc.hs [hunk 3]
```diff
@@ -157,37 +166,38 @@ instance HasConfiguration => Arbitrary SscPayloadDependsOnSlot where
...
+            rest <- arbitrary
+            pure (mkCommitmentsMap (map (genValidComm slot pm) rest))
+        genValidComm SlotId{..} pm (sk, c) = mkSignedCommitment pm sk siEpoch c
+
+        genVssCerts slot = do
+            pm <- arbitrary
+            rest <- arbitrary
+            pure (mkVssCertificatesMapLossy (map (genValidCert slot pm) rest))
+        genValidCert SlotId{..} pm (sk, pk) = mkVssCertificate pm sk pk $ siEpoch + 5
+-}
```
Comment (erikd) [inline ssc/Pos/Arbitrary/Ssc.hs]: Commented code?

File: block/src/Pos/Arbitrary/Block.hs [hunk 4]
```diff
@@ -145,17 +137,18 @@ instance (HasConfiguration, Arbitrary SscProof) => Arbitrary T.MainToSign where
...
+    arbitrary = do
         txPayload   <- arbitrary
-        generator   <- genPayloadDependsOnSlot <$> arbitrary
-        mpcData     <- generator slotId
-        dlgPayload  <- genDlgPayload $ Core.siEpoch slotId
+        mpcData     <- arbitrary
+        dlgPayload  <- arbitrary -- genDlgPayload $ Core.siEpoch slotId
         mpcUpload   <- arbitrary
         return $ T.MainBody txPayload mpcData dlgPayload mpcUpload
+-}
```
Comment (erikd) [inline block/src/Pos/Arbitrary/Block.hs]: Seems this is related to the changes in the `ssc` module.

**Comment by avieth** [comment]
Oh no it appears there are some stray intermediate changes that you've noticed. They're fixed up in the next one https://github.com/input-output-hk/cardano-sl/pull/2689

> I can definitely see the merit in these changes, but the PR is huge. I'm currently going through each separate commit in split diff mode. Its the only way to make sense of what is going on, but its time consuming.

Added you as a reviewer on this and on #2689. Do leave any and all questions here, this is still fresh in my mind.

File: ssc/Pos/Arbitrary/Ssc.hs [hunk 5]
```diff
@@ -38,11 +41,9 @@ import           Pos.Util.QuickCheck.Arbitrary (Nonrepeating (..), sublistN)
 -- Types
 ----------------------------------------------------------------------------
 
-{-
 newtype SscPayloadDependsOnSlot = SscPayloadDependsOnSlot
```
Comment (erikd) [inline ssc/Pos/Arbitrary/Ssc.hs]: This is the commented out code I mentioned earlier. Happy to see it back in.

File: lib/test/Test.hs [hunk 6]
```diff
@@ -3,6 +3,7 @@ import           Universum
 import           Test.Hspec (hspec)
 
 import           Spec (spec)
+
```
Comment (erikd) [inline lib/test/Test.hs]: The linter needed a blank line there? Really? :)

**Comment by erikd** [comment]
I'm ok, with this as it is. I can see it works towards the goal of reducing unneeded inter-dependencies and I think that's worthwhile.

**Comment by erikd** [comment]
Hmm, I don't seem to be able to approve this. Maybe because its already been merged into the `networking/develop` branch.

**Pull request merged**